### ISSUE TITLE
[2.28.0] panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by M-srivatsa** [prometheus/prometheus#11442]
### What did you do?

Prometheus started crashing without any change to config being done 

### What did you see instead? Under which circumstances?

Prometheus started crashing

### System information

Linux 4.15.0-1113-azure x86_64

### Prometheus version

```text
prometheus, version 2.28.0 (branch: HEAD, revision: ff58416a0b0224bab1f38f949f7d7c2a0f658940)
  build user:       root@32b9079a2740
  build date:       20210621-15:45:36
  go version:       go1.16.5
  platform:         linux/amd64
```

### Prometheus configuration file

```yaml
# my global config
global:
  scrape_interval:     1m # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 1m # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).

  # Attach these labels to any time series or alerts when communicating with
  # external systems (federation, remote storage, Alertmanager).
  external_labels:
      monitor: 'prometheus-monitor'

# read old data from prometheus v1
remote_read:
  - url: "http://localhost:9094/api/v1/read"

# Alertmanager service discovery
alerting:
  alertmanagers:
  - static_configs:
    - targets:
      - localhost:9093

# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  - "lms-recording.rules.yml"
  - "kafka-recording.rules.yml"
  - "recording.rules.yml"
  - "alert.rules.yml"
  - "kafka-alert.rules.yml"
  - "mysql-recording.rules.yml"

# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets: ['localhost:9090']
  - job_name: 'uptime'
    scrape_interval: 30s
    scheme: http
    static_configs:
      - targets: ['status.test.com']
    basic_auth: # Only needed if authentication is enabled (default)
      username: admin
      password: admin123
  - job_name: 'login-node'
    file_sd_configs:
      - files:
        - login-node.yml
  - job_name: 'live-node'
    file_sd_configs:
      - files:
        - live-node.yml
  - job_name: 'lite-node'
    file_sd_configs:
      - files:
        - lite-node.yml
  - job_name: 'bulk-node'
    file_sd_configs:
      - files:
        - bulk-node.yml
  - job_name: 'metrics-rest-node'
    file_sd_configs:
      - files:
        - metrics-rest-node.yml
  - job_name: 'metrics-batch-node'
    file_sd_configs:
      - files:
        - metrics-batch-node.yml
  - job_name: 'metrics-etl-node'
    file_sd_configs:
      - files:
        - metrics-etl-node.yml
  - job_name: 'kafka'
    static_configs:
      - targets: ['localhost:5556','localhost:5557','localhost:5558']
    relabel_configs:
      - source_labels: [__address__]
        regex: (localhost:5556)
        target_label: instance
        replacement: "TEST-VM-KAFKA01"
      - source_labels: [__address__]
        regex: (localhost:5557)
        target_label: instance
        replacement: "TEST-VM-KAFKA02"
      - source_labels: [__address__]
        regex: (localhost:5558)
        target_label: instance
        replacement: "TEST-VM-KAFKA03"
  - job_name: 'zoo'
    static_configs:
      - targets: ['localhost:5566','localhost:5567','localhost:5568']
    relabel_configs:
      - source_labels: [__address__]
        regex: (localhost:5566)
        target_label: instance
        replacement: "TEST-VM-ZOOKEEPER01"
      - source_labels: [__address__]
        regex: (localhost:5567)
        target_label: instance
        replacement: "TEST-VM-ZOOKEEPER02"
      - source_labels: [__address__]
        regex: (localhost:5568)
        target_label: instance
        replacement: "TEST-VM-ZOOKEEPER03"
  - job_name: 'kafka-consumer-lag'
    file_sd_configs:
      - files:
        - kafka-node.yml
  - job_name: 'mongo'
    file_sd_configs:
      - files:
        - mongo-node.yml
  - job_name: 'elasticsearch'
    file_sd_configs:
      - files:
        - es-node.yml
  - job_name: 'mysql'
    file_sd_configs:
      - files:
        - mysql-node.yml
  - job_name: 'node'
    azure_sd_configs:
    - subscription_id: "****************************"
      authentication_method: ManagedIdentity
      environment: AzurePublicCloud
      refresh_interval: 120s
      port: 9100
    relabel_configs:
      - source_labels: [__meta_azure_machine_name]
        target_label: instance
        regex: '(ACI-PROD-|VM-)(.+)'
        replacement: '$1$2:9100'
      - source_labels: [__address__]
        regex: (TEST-VM-LOGIN).+:9100
        target_label: server_role
        replacement: "appserver"
      - source_labels: [__address__]
        regex: (TEST-VM-LMS).+:9100
        target_label: server_role
        replacement: "appserver"
      - source_labels: [__address__]
        regex: (TEST-VM-BULK).+:9100
        target_label: server_role
        replacement: "appserver"
      - source_labels: [__address__]
        regex: (TEST-VM-CRON).+:9100
        target_label: server_role
        replacement: "appserver"
      - source_labels: [__address__]
        regex: (TEST-VM-ELASTIC).+:9100
        target_label: server_role
        replacement: "elasticsearch"
      - source_labels: [__address__]
        regex: (TEST-VM-MONGOC).+:9100
        target_label: server_role
        replacement: "mongo"
      - source_labels: [__address__]
        regex: (TEST-VM-MYSQL).+:9100
        target_label: server_role
        replacement: "mysql"
      - source_labels: [__address__]
        regex: (TEST-VM-CRM).+:9100
        target_label: server_role
        replacement: "crm"
      - source_labels: [__address__]
        regex: (TEST-VM-TARGET).+:9100
        target_label: server_role
        replacement: "target"
      - source_labels: [__address__]
        regex: (TEST-VM-ZOOKEEPER).+:9100
        target_label: server_role
        replacement: "zookeeper"
      - source_labels: [__address__]
        regex: (TEST-VM-KAFKA).+:9100
        target_label: server_role
        replacement: "kafka"
      - source_labels: [__address__]
        regex: (TEST-VM-RP).+:9100
        target_label: server_role
        replacement: "webserver"
      - source_labels: [__address__]
        regex: (TEST-VM-BASTION).+:9100
        target_label: server_role
        replacement: "bastion"
      - source_labels: [__address__]
        regex: (TEST-VM-ETL).+:9100
        target_label: server_role
        replacement: "metrics-etl"
      - source_labels: [__address__]
        regex: (TEST-VM-BATCH).+:9100
        target_label: server_role
        replacement: "metrics-batch"
      - source_labels: [__address__]
        regex: (TEST-VM-REST).+:9100
        target_label: server_role
        replacement: "metrics-rest"
      - source_labels: [__address__]
        regex: (TEST-VM-INTELLICONTROLLER).+:9100
        target_label: server_role
        replacement: "intelli-controller"
      - source_labels: [__address__]
        regex: (TEST-VM-INTELLIWORKER).+:9100
        target_label: server_role
        replacement: "intelli-worker"
      - source_labels: [__address__]
        regex: (TEST-VM-OHS).+:9100
        target_label: server_role
        replacement: "org-hierarchy-service"
      - source_labels: [__address__]
        regex: (TEST-VM-GEOETL).+:9100
        target_label: server_role
        replacement: "geo-etl"
      - source_labels: [__address__]
        regex: (TEST-VM-GEOBATCH).+:9100
        target_label: server_role
        replacement: "geo-batch"
```

### Logs

```text
level=info ts=2022-10-10T19:14:55.121Z caller=head.go:854 component=tsdb msg="WAL segment loaded" segment=17816 maxSegment=17818
level=info ts=2022-10-10T19:14:55.295Z caller=head.go:854 component=tsdb msg="WAL segment loaded" segment=17817 maxSegment=17818
level=info ts=2022-10-10T19:14:55.295Z caller=head.go:854 component=tsdb msg="WAL segment loaded" segment=17818 maxSegment=17818
level=info ts=2022-10-10T19:14:55.295Z caller=head.go:860 component=tsdb msg="WAL replay completed" checkpoint_replay_duration=6.690868306s wal_replay_duration=2.224069515s total_replay_duration=8.92022778s
level=info ts=2022-10-10T19:14:58.992Z caller=main.go:834 fs_type=EXT4_SUPER_MAGIC
level=info ts=2022-10-10T19:14:58.992Z caller=main.go:837 msg="TSDB started"
level=info ts=2022-10-10T19:14:58.992Z caller=main.go:964 msg="Loading configuration file" filename=/opt/prometheus/prometheus-2.28.0.linux-amd64/prometheus.yml
level=info ts=2022-10-10T19:14:59.011Z caller=main.go:995 msg="Completed loading of configuration file" filename=/opt/prometheus/prometheus-2.28.0.linux-amd64/prometheus.yml totalDuration=19.253879ms remote_storage=86.302µs web_handler=700ns query_engine=1µs scrape=855.826µs scrape_sd=382.312µs notify=18.6µs notify_sd=17.301µs rules=16.340091ms
level=info ts=2022-10-10T19:14:59.011Z caller=main.go:779 msg="Server is ready to receive web requests."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xfe4a1e]

goroutine 2168 [running]:
github.com/prometheus/prometheus/discovery/azure.(*Discovery).refresh.func1(0xc04823d1f0, 0xc08005d3e0, 0xc054d3e480, 0xc021520b00, 0x37ae490, 0xc0001b7580, 0x125, 0xc03aaf00a0, 0x92, 0xc036e29fd0, ...)
	/app/discovery/azure/azure.go:350 +0xb5e
created by github.com/prometheus/prometheus/discovery/azure.(*Discovery).refresh
	/app/discovery/azure/azure.go:296 +0x645
```

**Comment by tjucoder** [comment]
Maybe need check ip.PublicIPAddress.IPAddress nil here :)

See: https://github.com/prometheus/prometheus/commit/002ceb723299286b943185486ff765b6cef999d3

**Comment by beorn7** [comment]
Hello from the bug scrub!

We believe this has been fixed indeed by #10289. Closing.
